processors/metricremover/metricremover.go

```diff
@@ -0,0 +1,27 @@
+package metricremover
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.uber.org/zap"
+)
+
+type processor struct {
+	removeNoneMetricType bool
+	logger               *zap.Logger
+}
+
+// ProcessMetrics implements processorhelper.ProcessMetricsFunc
+func (p *processor) ProcessMetrics(ctx context.Context, metrics pmetric.Metrics) (pmetric.Metrics, error) {
+	rms := metrics.ResourceMetrics()
```

Comment:
would it be worth doing the `removeNoneMetricType` check above the for loop so we don't unnecessarily iterate over all metrics?

```
func processMetrics()
  if p.removeNoneMetricType != true
    return metrics, nil
  end
  
  ... for loop to iterate over metrics & delete if they are none type
  ```

Reply:
ah good eye :)